app/src/main/java/me/guillaumin/android/osmtracker/gpx/TrackFileProvider.java

```diff
@@ -0,0 +1,121 @@
+package me.guillaumin.android.osmtracker.gpx;
+
+import android.content.ContentProvider;
+import android.content.ContentValues;
+import android.database.AbstractCursor;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.ParcelFileDescriptor;
+import android.util.Log;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+
+/**
```

Comment:
I think the comment would be useful here.

It's been a while since I've used `ContentProvider`s, but I assume the only thing that matter in this one is `getTypes()` and `getStreamTypes()` since the other methods shouldn't be called by the 3rd party app the file is shared to (query, insert, delete, ...)?

If that's the case, it may be preferable to throw exceptions in these methods rather than returning default values like "0", as I suspect it may lead into hard to diagnose issue if someone tries to use this content provider in a different way. Perhaps something like `throw new IllegalStateException("This method cannot be used with this content provider")`?